src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -676,6 +1230,18 @@ void AtlasEngine::_createResources()
             THROW_IF_FAILED(_r.device->CreateBuffer(&desc, nullptr, _r.constantBuffer.put()));
         }
 
+        if (!_api.customPixelShaderImagePath.empty())
+        {
+            _r.customShaderTexture = LoadCustomShaderTexture(
+                _r.device.get(),
+                _r.deviceContext.get(),
+                _api.customPixelShaderImagePath);
+        }
+        else
+        {
+            _r.customShaderTexture = CustomShaderTexture();
+        }

```

Comment:
You can drop this else branch. `_r` will always be emptied initially.

Reply:
Done